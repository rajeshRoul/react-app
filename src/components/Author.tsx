import React, { ReactElement } from 'react'
import '../styles/Author.css'

function Author(props: { url: string; name: string }): ReactElement {
    const { url, name } = props
    return (
        <div className="Author">
            <img src={url} alt={name} />
            <h2>{name}</h2>
        </div>
    )
}

export default Author
