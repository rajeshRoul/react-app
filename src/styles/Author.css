.Author{
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 30px 10px 0;
}

.Author img{
    width: 450px;
    height: 350px;
}

@media screen and (max-width: 460px) {
    .Author img{
        width: 100%;
        height: auto;
    }
}