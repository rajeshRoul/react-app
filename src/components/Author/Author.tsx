import React, { ReactElement } from 'react'
import './Author.css'

function Author(props: { url: string; name: string }): ReactElement {
    const { url, name } = props
    return (
        <div data-testid="author" className="Author">
            <img src={url} alt={name} />
            <h2 data-testid="authorName">{name}</h2>
        </div>
    )
}

export default Author
