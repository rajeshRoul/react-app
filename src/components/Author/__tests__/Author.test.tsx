/* eslint-disable no-undef */
import React from 'react'
import { render, cleanup, screen } from '@testing-library/react'
import Author from '../Author'
import '@testing-library/jest-dom/extend-expect'

afterEach(cleanup)

test('renders Author component', () => {
    const name = 'test'
    const url = '/test'
    const { getByTestId } = render(<Author name={name} url={url} />)
    expect(getByTestId('authorName')).toHaveTextContent(name)
    const imgElement = screen.getByAltText(name)
    expect(imgElement).toBeInTheDocument()
})

it('should take a snapshot', () => {
    const name = 'test'
    const url = '/test'
    const { getByTestId } = render(<Author name={name} url={url} />)
    expect(getByTestId('author')).toMatchSnapshot()
})
