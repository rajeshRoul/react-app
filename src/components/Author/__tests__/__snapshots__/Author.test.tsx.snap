// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div
  class="Author"
  data-testid="author"
>
  <img
    alt="test"
    src="/test"
  />
  <h2
    data-testid="authorName"
  >
    test
  </h2>
</div>
`;
